zio-http-example/src/main/scala/example/codecs/RequestBodyJsonDeserializationExample.scala

```diff
@@ -0,0 +1,37 @@
+package example.codecs
+
+import zio._
+
+import zio.schema.codec.JsonCodec.schemaBasedBinaryCodec
+import zio.schema.{DeriveSchema, Schema}
+
+import zio.http._
+
+object RequestBodyJsonDeserializationExample extends ZIOAppDefault {
+
+  case class Book(title: String, authors: List[String])
+
+  object Book {
+    implicit val schema: Schema[Book] = DeriveSchema.gen
+  }
+
+  val app: Routes[Ref[List[Book]], Nothing] =
+    Routes(
+      Method.POST / "books" ->
+        handler { (req: Request) =>
+          for {
+            book  <- req.body.to[Book].catchAll(_ => ZIO.fail(Response.badRequest("unable to deserialize the request")))
+            books <- ZIO.service[Ref[List[Book]]]
+            _     <- books.updateAndGet(_.appended(book))
```

Comment:
try instead `_  :+ book`

Reply:
Thank you! Fixed!
I'm curious, what is the difference between `:+` and `appended`?